# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/WebrtcGmpVideoCodec.h
# Commit: 9c021d42060a
# Full Hash: 9c021d42060a464f5c1acbf917d8280a569dc6e9
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2025-03-14 04:45:14
# Description:
#   Bug 1953722 - Key input frames to the webrtc gmp encoder off ntp timestamps instead of rtp. r=aosmond
#   
#   This to avoid time going backwards due to uint32_t wraparound.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D241394
# ==============================================================================

diff -r d0045739b9f8 -r 9c021d42060a dom/media/webrtc/libwebrtcglue/WebrtcGmpVideoCodec.h
--- a/dom/media/webrtc/libwebrtcglue/WebrtcGmpVideoCodec.h	Thu Mar 13 15:11:44 2025 +0000
+++ b/dom/media/webrtc/libwebrtcglue/WebrtcGmpVideoCodec.h	Thu Mar 13 15:25:10 2025 +0000
@@ -58,8 +58,10 @@
 
 namespace mozilla::detail {
 struct InputImageData {
-  uint64_t rtp_timestamp = 0;
+  uint64_t gmp_timestamp_us = 0;
+  int64_t ntp_timestamp_ms = 0;
   int64_t timestamp_us = 0;
+  uint32_t rtp_timestamp = 0;
   webrtc::ScalableVideoController::LayerFrameConfig frame_config;
 };
 }  // namespace mozilla::detail
